src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -116,5 +122,38 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
 
             return builder;
         }
+
+        private static void UseHealthChecks(IApplicationBuilder app)
+        {
+            // '/runtime' is a reserved API path. We use that to avoid conflict with any customer function routes.
+            const string healthPrefix = "/runtime/health";
+            static bool Predicate(HttpContext context)
+            {
+                return context.Request.Path.StartsWithSegments(healthPrefix);
+            }
+
+            app.MapWhen(Predicate, app =>
+            {
+                // This supports the ?wait={seconds} query string.
+                app.UseMiddleware<HealthCheckWaitMiddleware>();
+
+                app.UseHealthChecks(healthPrefix, new HealthCheckOptions
+                {
+                    ResponseWriter = HealthCheckResponseWriter.WriteResponseAsync,
+                });
+
+                app.UseHealthChecks($"{healthPrefix}/live", new HealthCheckOptions
```

Comment:
So this is the line that is actually registering the http route on the app? What are the auth requirements for these endpoints? Because it looks like in some cases, e.g. [here](https://github.com/Azure/azure-functions-host/blob/efbc6b12326b4cb041880336d4851cd2233f26d3/src/WebJobs.Script/Diagnostics/HealthChecks/ScriptHostHealthCheck.cs#L47) we're returning app details.